virtcontainers/kata_agent.go

```diff
@@ -1253,6 +1254,40 @@ func (k *kataAgent) appendVhostUserBlkDevice(dev ContainerDevice, c *Container)
 	return kataDevice
 }
 
+func (k *kataAgent) appendVfioDevice(dev ContainerDevice, c *Container) *grpc.Device {
+	device := c.sandbox.devManager.GetDeviceByID(dev.ID)
```

Comment:
This function can return `nil`, but you are deferencing the return value below.

Reply:
That also appears to be the case for all the existing `append*Device` functions.

Reply:
Ok. Not strictly required for this PR, but it would be great if you could add a `nil` check for all the handlers on a separate commit.

Reply:
It turns out the caller has already nil checked this value, though it's not very obvious here.  I'm making use of that for my 2.x version of this series.